omniscidb/ResultSet/RowSetMemoryOwner.cpp

```diff
@@ -43,6 +44,11 @@ quantile::TDigest* RowSetMemoryOwner::nullTDigest(double const q) {
       .get();
 }
 
+hdk::quantile::Quantile* RowSetMemoryOwner::quantile() {
+  std::lock_guard<std::mutex> lock(state_mutex_);
+  return quantiles_.emplace_back(std::make_unique<hdk::quantile::Quantile>(this)).get();
```

Comment:
We keep all allocations in RowSetMemoryOwner thread-safe and the lock is required because `quantiles_` vector is modified. I'm not sure if we ever use the same memory owner from multiple threads for allocations though. Here I just keep things as they used to be. We might state that it doesn't have to be thread-safe and remove locks here and in other similar places.

Reply:
I wasn't sure as well, hence the question. Anyway, this is a separate matter.